File: Selenium/Project/task5.java
package project;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

import java.awt.*;

import static org.testng.AssertJUnit.assertTrue;

public class task5 {
    WebDriver driver;
    // Navigate to another page
    // Goal: Navigate to the “My Account” page on the site.

    @BeforeClass
    public void beforeMethod() {
        // Set up the firefox driver
        WebDriverManager.firefoxdriver().setup();
        // Create a new instance of the Firefox driver
        driver = new FirefoxDriver();

        // Open a browser.
        // Navigate to ‘https://alchemy.hguy.co/lms’
        driver.get("https://alchemy.hguy.co/lms");
    }

    @Test
    public void clickMyAccount() {

        //  Find the navigation bar
        // Select the menu item that says “My Account” and click it
        WebElement navigationBar = driver.findElement(By.linkText("My Account"));
        String navigatorBarText = navigationBar.getText();
        System.out.println("Navigation bar text is: " + navigatorBarText);
        navigationBar.click();

        // Read the page title and verify that you are on the correct page
        String currentPageTitle = driver.getTitle();
        System.out.println("Current page Title is: " + currentPageTitle);
        assertTrue(currentPageTitle.contains(navigatorBarText));
    }

    @AfterClass
    public void afterMethod() {
        //  Close the browser
        driver.quit();
    }
}

File: Selenium/Project/task5.py
# Import webdriver from selenium
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager

# Navigate to another page
# Goal: Navigate to the “My Account” page on the site.


# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Navigate to ‘https://alchemy.hguy.co/lms’.
    driver.get("https://alchemy.hguy.co/lms/")

    # Find the navigation bar.
    # Select the menu item that says "My Account" and click it.
    navigationBar = driver.find_element(By.XPATH,"//*[@id='menu-item-1507']/a")
    navigationBarText = navigationBar.text
    navigationBar.click()
    print("Title of New webpage is:", navigationBarText)

    # Read the page title and verify that you are on the correct page
    currentPageTitle = driver.title
    print("Title of new page title is: ", currentPageTitle)
    assert navigationBarText in currentPageTitle

    # If it matches, close the browser
    driver.quit